streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -411,14 +413,18 @@ void closeSuspended(boolean clean, RuntimeException firstException) {
             if (eosEnabled) {
                 if (!clean) {
                     try {
-                        producer.abortTransaction();
+                        if (!isZombie) {
+                            producer.abortTransaction();
+                        }
                         transactionInFlight = false;
                     } catch (final ProducerFencedException e) {
                         // can be ignored: transaction got already aborted by brokers/transactional-coordinator if this happens
                     }
                 }
                 try {
-                    recordCollector.close();
+                    if (!isZombie) {
```

Comment:
Because record collector calls `Producer.close()` and this will result in an exception -- after a  producer got fenced, you should not call any method anymore.